File: src/main/scala/com/quine/language/server/QuineLanguageServer.scala
//package com.quine.language.server
//
//import org.eclipse.lsp4j.{InitializeParams, InitializeResult, InitializedParams, SetTraceParams, WorkDoneProgressCancelParams}
//import org.eclipse.lsp4j.services.{LanguageServer, NotebookDocumentService, TextDocumentService, WorkspaceService}
//
//import java.util.concurrent.CompletableFuture
//
//class QuineLanguageServer extends LanguageServer {
//  override def initialized(params: InitializedParams): Unit = ???
//
//  override def initialized(): Unit = ???
//
//  override def getNotebookDocumentService: NotebookDocumentService = super.getNotebookDocumentService
//
//  override def cancelProgress(params: WorkDoneProgressCancelParams): Unit = ???
//
//  override def setTrace(params: SetTraceParams): Unit = ???
//
//  override def initialize(params: InitializeParams): CompletableFuture[InitializeResult] = ???
//
//  override def shutdown(): CompletableFuture[AnyRef] = ???
//
//  override def exit(): Unit = ???
//
//  override def getTextDocumentService: TextDocumentService = new QuineTextDocumentService
//
//  override def getWorkspaceService: WorkspaceService = ???
//}

File: src/main/java/com/quine/language/server/QuineLanguageServer.java
package com.quine.language.server;

import org.eclipse.lsp4j.InitializeParams;
import org.eclipse.lsp4j.InitializeResult;
import org.eclipse.lsp4j.services.LanguageServer;
import org.eclipse.lsp4j.services.TextDocumentService;
import org.eclipse.lsp4j.services.WorkspaceService;

import java.util.concurrent.CompletableFuture;

public class QuineLanguageServer implements LanguageServer {
    @Override
    public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
        return null;
    }

    @Override
    public CompletableFuture<Object> shutdown() {
        return null;
    }

    @Override
    public void exit() {

    }

    @Override
    public TextDocumentService getTextDocumentService() {
        return new QuineTextDocumentService();
    }

    @Override
    public WorkspaceService getWorkspaceService() {
        return null;
    }
}
